File: polyglot/Singleton/singleton.hs
-- Singleton design pattern

data SingleObject = SingleObject deriving Eq

getInstance :: SingleObject
getInstance = SingleObject

showMessage :: SingleObject -> IO ()
showMessage SingleObject = putStrLn "The single object is alive and well"

main = do
  let object1 = getInstance in
    showMessage object1 >>
    let object2 = getInstance in
      if object1 == object2 
        then putStrLn "The two objects are the same"
        else return ()

File: polyglot/Singleton/singleton.js
// Singleton design pattern
function SingleObject(){

};
SingleObject.prototype.showMessage = function(){
   print("The single object is alive and well")
};
SingleObject.getInstance =function(){
  return SingleObject.prototype;
}


object1 = SingleObject.getInstance();
object1.showMessage();

object2 = SingleObject.getInstance();
if(object1 === object2){
  print("The two objects are the same");
}

